File: src/main/java/br/com/estimular/atendimento/model/converters/SexoConverter.java
package br.com.estimular.atendimento.model.converters;

import br.com.estimular.atendimento.model.enums.Sexo;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class SexoConverter implements AttributeConverter<Sexo, String> {

    @Override
    public String convertToDatabaseColumn(Sexo attribute) {
        if (attribute == null) {
            return null;
        }
        return attribute.getCodigo();
    }

    @Override
    public Sexo convertToEntityAttribute(String dbData) {
        if (dbData == null) {
            return null;
        }
        return Sexo.fromCodigo(dbData);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/converters/SexoConverter.kt
package com.clinica.atendimento.model.converters

import com.clinica.atendimento.model.enums.Sexo
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class SexoConverter : AttributeConverter<Sexo, String> {

    override fun convertToDatabaseColumn(attribute: Sexo?): String? {
        return attribute?.codigo
    }

    override fun convertToEntityAttribute(dbData: String?): Sexo? {
        return if (dbData == null) null else Sexo.fromCodigo(dbData)
    }
}